### ISSUE TITLE
Support for async tests?

### ISSUE THREAD

**Issue by togakangaroo** [approvals/ApprovalTests.Python#114]
This is using pytest_asyncio with [the pytest plugin](https://github.com/approvals/ApprovalTests.Python.PytestPlugin)

When I have this
```
@pytest.mark.asyncio
async def test__can_approve_sql():
    verify("Foo")
```

I get the following error

```

    @pytest.mark.asyncio
    async def test__can_approve_sql():
>       verify("Foo")

tests/test_forms_api.py:381: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.venv/lib/python3.10/site-packages/approvaltests/core/options.py:65: in namer
    namer = self.fields.get("namer", get_default_namer())
.venv/lib/python3.10/site-packages/approvaltests/namer/default_name.py:8: in get_default_namer
    return StackFrameNamer(extension)
.venv/lib/python3.10/site-packages/approvaltests/namer/stack_frame_namer.py:17: in __init__
    self.set_for_stack(inspect.stack(1))
.venv/lib/python3.10/site-packages/approvaltests/namer/stack_frame_namer.py:22: in set_for_stack
    frame = self.get_test_frame(caller)
.venv/lib/python3.10/site-packages/approvaltests/namer/stack_frame_namer.py:39: in get_test_frame
    if self.is_test_method(frame):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

frame = FrameInfo(frame=<frame at 0x10e68aae0, file '/Users/gmauer/.pyenv/versions/3.10.4/lib/python3.10/asyncio/events.py', l...y', lineno=80, function='_run', code_context=['            self._context.run(self._callback, *self._args)\n'], index=0)

    @staticmethod
    def is_test_method(frame: FrameInfo) -> bool:
        method_name = frame[3]
        is_unittest_test = (
            "self" in frame[0].f_locals
>           and "_testMethodName" in frame[0].f_locals["self"].__dict__
            and method_name != "__call__"
            and method_name != "_callTestMethod"
            and method_name != "run"
        )
E       AttributeError: 'Handle' object has no attribute '__dict__'

.venv/lib/python3.10/site-packages/approvaltests/namer/stack_frame_namer.py:53: AttributeError
============================================================================ short test summary info =============================================================================
FAILED tests/test_forms_api.py::test__can_approve_sql - AttributeError: 'Handle' object has no attribute '__dict__'
========================================================================== 1 failed, 2 passed in 1.83s ===========================================================================

```

The error is reasonable when this is not async

**Comment by isidore** [comment]
We should do this, would you be able to join us one week for the mob to make sure it goes correctly:
[Weekly Mob Details here](https://github.com/approvals/ApprovalTests.Python/blob/main/docs/Contribute.md#join-our-weekly-mobbing-sessions)

**Comment by togakangaroo** [comment]
@isidore That is the one time that I actually probably *can* join. Likely not this, but the following week, and I would be thrilled to join one of your mobbing sessions.

**Comment by isidore** [comment]
Solved in Release 5.4.2
